File: languages/ruby/test-one.rb
#!/usr/bin/env ruby

require 'bundler/setup'
require 'microformats'
require 'json'

def print_usage
  puts 'Usage: microformats (URL, filepath, or HTML)'
end

def process_html(file)
  baseUrl = 'http://example.com/'
  if file.include?('/microformats-v2-unit/')
    # This is a unit test; these use a different base URL
    baseURL = 'http://example.test'
  end
  collection = Microformats.parse(file, base: baseUrl)
  puts JSON.pretty_generate(JSON[collection.to_json.to_s])
end

if ARGV[0].nil?
  print_usage
else
  process_html(ARGV[0])
end

File: languages/elixir/test-one.exs
defmodule MicroformatsTester do
  def main do
    if length(System.argv()) != 1 do
        IO.puts("Usage: test_one <inputfile>")
        exit(1)
    end
    file_name = Enum.at(System.argv(), 0)

    base_url = if String.contains?(file_name, "/microformats-v2-unit"), do: "http://example.test/", else: "http://example.com/"

    html = File.read!(file_name)
    out = JSON.encode!(Microformats2.parse(html, base_url))

    IO.puts(out)
  end
end

MicroformatsTester.main()
